### PR TITLE
포스트 자동저장 실패 시 토스트 피드백

### PR DESCRIPTION
**Pull request by robin-maki**
근데 아무리 임시라지만 정말 이렇게 짜도 되는 걸까요...

### CODE REVIEW

**Comment by robin-maki** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/penxle/penxle/1394?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#1394** <a href="https://app.graphite.dev/github/pr/penxle/penxle/1394?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1384** <a href="https://app.graphite.dev/github/pr/penxle/penxle/1384?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1376** <a href="https://app.graphite.dev/github/pr/penxle/penxle/1376?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/penxle/penxle/1394?utm_source=stack-comment).

File: apps/penxle.com/src/routes/editor/[permalink]/Editor.svelte [hunk 1]
```diff
@@ -86,26 +87,30 @@
...
+      try {
+        const resp = await revisePost({
+          revisionKind,
+          postId: $post.id,
+          title: $store.title,
+          subtitle: $store.subtitle,
+          content: $store.content,
+          paragraphIndent: $store.paragraphIndent,
+          paragraphSpacing: $store.paragraphSpacing,
+        });
```
Comment (x86chi) [inline apps/penxle.com/src/routes/editor/[permalink]/Editor.svelte:99]: P3: mixpanel.track 까지 포괄하고 있어 mixpanel tracking 이 실패해도 오류 토스트가 띄워질텐데 revisePost 으로 좁히는게 어떨까요?

File: apps/penxle.com/src/routes/editor/[permalink]/Editor.svelte [hunk 2]
```diff
@@ -86,15 +87,21 @@
...
-      const resp = await revisePost({
-        revisionKind,
-        postId: $post.id,
-        title: $store.title,
-        subtitle: $store.subtitle,
-        content: $store.content,
-        paragraphIndent: $store.paragraphIndent,
-        paragraphSpacing: $store.paragraphSpacing,
-      });
+      let resp;
```
Comment (x86chi) [inline apps/penxle.com/src/routes/editor/[permalink]/Editor.svelte:90]:
```suggestion
      let resp: Awaited<ReturnType<typeof revisePost>> | undefined;
      try {
        resp = await revisePost({
      }
      // 중략
      if (!resp) throw new Error('No response');
```

P2: `resp` 타입 명시 부탁드립니다!

**Review by x86chi — CHANGES_REQUESTED** [review]

File: apps/penxle.com/src/routes/editor/[permalink]/Editor.svelte [hunk 3]
```diff
@@ -88,26 +89,31 @@
...
+        $state.lastRevision = {
+          kind: resp.draftRevision.kind,
+          updatedAt: resp.draftRevision.updatedAt,
+        };
+
+        warnUnload = false;
+
+        return resp.draftRevision.id;
+      } catch (err) {
+        toast.error('포스트를 저장하는 중에 문제가 발생했어요');
```
Comment (x86chi) [inline apps/penxle.com/src/routes/editor/[permalink]/Editor.svelte:114]: revisePost muation 오류 응답에 오류 ID가 있으면 해당 ID 를 함께 보여주면 어떨까 싶은데요, @devunt  해당 정보를 알 수 있나요?

**Pull request closed without merging**